### PR TITLE
ci(): Refactor GHA for reuse and caching

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

I was trying to remove the extra run of the tests for coverage and i went in a rabbit hole with caching.

## Changes

- added a cached action for the install
- added a cached action to build fabric ( building fabric is actually very slow in ci )
- pulled out from the job matrix node 18 and made its own job with coverage
- added node 20 in the tests

There are some improvements:

This is a standar node test run now:
image

While before:
image

The build part that was taking 32 seconds is gone, while the cached install is taking anywat the same amount of before, meaning that either NPM is super fast, or that the cache is unefficient.

I do prefer to have it cached since at least the bandwidth is internal rather than external

### CODE REVIEW

**Comment by CODE-AXION** [comment]
@asturur  hey did you checked by PR ? https://github.com/fabricjs/fabric.js/pull/8955

**Comment by asturur** [comment]
@CODE-AXION this is not how it works.
You can't jump on related PRs bumping up your.
We can check it when we can ckeck it. We are trying to prioritize work to get 6.0 published.

**Comment by asturur** [comment]
ok this starts to be good.

**Comment by asturur** [comment]
I think i m done with this.
The job prime-build is something i m not entirely sold into yet, but i would like to leave it on for a bit.
It makes everything start later, but it avoids building fabricJS more than once for the entire test run.
So we wait a bit more, but on the pipeline we save 11 builds after.

If we let every job start in parallel most o them will get to build time when the build is not cached yet.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -18,18 +60,19 @@ jobs:
...
+      - uses: ./.github/actions/cached-install
         with:
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run build -- -f
+      - name: Build fabric.js
+        uses: ./.github/actions/build-fabric-cached
       - name: Run ${{ matrix.target }} ${{ matrix.suite }} headless test
-        uses: GabrielBB/xvfb-action@v1
+        uses: coactions/setup-xvfb@v1
```
[the remark this answers is not in the document]
Comment (ShaMan123) [reply .github/workflows/tests.yml:71]: what about this?

File: .github/actions/build-fabric-cached/action.yml
Comment (ShaMan123) [inline .github/actions/build-fabric-cached/action.yml:1]: So this is like a function?
I should turn the changelog actions into this.

Comment (asturur) [reply .github/actions/build-fabric-cached/action.yml:1]: is an action made of more steps that you don't need to publish. If is not reusable outside fabricJS is a good idea to have it here

File: .github/workflows/tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (asturur) [reply .github/workflows/tests.yml:71]: even better imho. I forgot about the visual artifact prs, we should merge that too.

**Pull request merged**